File: data/recomposed_projects/llama-3-70b-instruct/body/commons-codec/evosuite-test/src/test/org/apache/commons/codec/language/Nysiis_ESTest.py
from __future__ import annotations
import time
import re
import os
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.codec.language.Nysiis import *

# from src.test.org.apache.commons.codec.language.Nysiis_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class Nysiis_ESTest(unittest.TestCase):

    def test25(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis("org.apache.commons.codec.EncoderException")
        self.assertEqual("ORGAPA", string0)
        self.assertIsNotNone(string0)

    def test24(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        object0 = nysiis0.encode0("c}=[ljGyI?xU<h4{Ui")
        self.assertEqual("CLJGYA", object0)
        self.assertIsNotNone(object0)

    def test23(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.encode1("By@EbheBevSR p2+n")
        self.assertEqual("BYABAB", string0)
        self.assertIsNotNone(string0)

    def test22(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        object0 = nysiis0.encode0("Ybqp-Hd(")
        self.assertTrue(nysiis0.isStrict())
        self.assertEqual("YBGFD", object0)

    def test21(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis(")-ZF*!\u0002'6Jjp}kNy")
        self.assertEqual("ZFJPNY", string0)
        self.assertTrue(nysiis0.isStrict())

    def test20(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.encode1("8caW")
        self.assertTrue(nysiis0.isStrict())
        self.assertEqual("C", string0)
        self.assertIsNotNone(string0)

    def test18(self) -> None:

        nysiis0 = Nysiis(False)
        nysiis0.nysiis(None)
        self.assertFalse(nysiis0.isStrict())

    def test17(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        nysiis0.encode1("")
        self.assertTrue(nysiis0.isStrict())

    def test15(self) -> None:

        nysiis0 = Nysiis(False)
        object0 = nysiis0.encode0("org.apache.commons.codec.EncoderException")
        self.assertEqual(object0, "ORGAPACACANANSCADACANCADARAXCAPTAN")

    def test14(self) -> None:
        nysiis0 = Nysiis.Nysiis1()
        boolean0 = nysiis0.isStrict()
        self.assertTrue(boolean0)

    def test13(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        try:
            nysiis0.encode0(nysiis0)
            self.fail("Expecting exception: Exception")

        except Exception as e:
            # Parameter supplied to Nysiis encode is not of type java.lang.String
            self.verifyException("org.apache.commons.codec.language.Nysiis", e)

    def test12(self) -> None:

        nysiis0 = Nysiis(False)
        nysiis0.nysiis("")
        self.assertFalse(nysiis0.isStrict())

    def test11(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        nysiis0.nysiis("N")
        self.assertTrue(nysiis0.isStrict())

    def test10(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis("T2ay`f=A)4z+>/-")
        self.assertEqual("TAYF", string0)
        self.assertTrue(nysiis0.isStrict())

    def test09(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis(":4d2],8DdU<]")
        self.assertTrue(nysiis0.isStrict())
        self.assertEqual("D", string0)

    def test08(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis("(EE|IE)$")
        self.assertEqual("EY", string0)
        self.assertTrue(nysiis0.isStrict())

    def test07(self) -> None:

        nysiis0 = Nysiis(False)
        string0 = nysiis0.nysiis(">X3+b]+v")
        self.assertEqual("XBV", string0)
        self.assertFalse(nysiis0.isStrict())

    def test04(self) -> None:

        nysiis0 = Nysiis(True)
        nysiis0.encode1(None)
        self.assertTrue(nysiis0.isStrict())

    def test03(self) -> None:

        nysiis0 = Nysiis(False)
        boolean0 = nysiis0.isStrict()
        self.assertFalse(boolean0)

    def test01(self) -> None:

        nysiis0 = Nysiis.Nysiis1()
        string0 = nysiis0.nysiis("RClpkggkVkz`Mr?-hN")
        self.assertEqual("RCLPCG", string0)

    def test00(self) -> None:

        nysiis0 = Nysiis(True)
        string0 = nysiis0.encode1("HSCACT")
        self.assertTrue(nysiis0.isStrict())
        self.assertEqual("HSCACT", string0)

File: java_projects/cleaned_final_projects_evosuite/commons-codec/evosuite-tests/org/apache/commons/codec/language/Nysiis_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.codec.language;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.codec.language.Nysiis;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class Nysiis_ESTest extends Nysiis_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(true);
      String string0 = nysiis0.encode1("HSCACT");
      assertTrue(nysiis0.isStrict());
      assertEquals("HSCACT", string0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis("RClpkggkVkz`Mr?-hN");
      assertEquals("RCLPCG", string0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.encode("m7l@s?b4kic>saDWeH");
      assertEquals("MLSBCA", string0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      boolean boolean0 = nysiis0.isStrict();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(true);
      nysiis0.encode1((String) null);
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      nysiis0.encode((String) null);
      assertFalse(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      nysiis0.encode("");
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      String string0 = nysiis0.nysiis(">X3+b]+v");
      assertEquals("XBV", string0);
      assertFalse(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis("(EE|IE)$");
      assertEquals("EY", string0);
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis(":4d2],8DdU<]");
      assertTrue(nysiis0.isStrict());
      assertEquals("D", string0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis("T2ay`f=A)4z+>/-");
      assertEquals("TAYF", string0);
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      nysiis0.nysiis("N");
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      nysiis0.nysiis("");
      assertFalse(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      try { 
        nysiis0.encode0(nysiis0);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Parameter supplied to Nysiis encode is not of type java.lang.String
         //
         verifyException("org.apache.commons.codec.language.Nysiis", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      boolean boolean0 = nysiis0.isStrict();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      Object object0 = nysiis0.encode0("org.apache.commons.codec.EncoderException");
      assertEquals("ORGAPACACANANSCADACANCADARAXCAPTAN", object0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      nysiis0.encode((Object) "D");
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      nysiis0.encode1("");
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      Nysiis nysiis0 = new Nysiis(false);
      nysiis0.nysiis((String) null);
      assertFalse(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      try { 
        nysiis0.encode((Object) null);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Parameter supplied to Nysiis encode is not of type java.lang.String
         //
         verifyException("org.apache.commons.codec.language.Nysiis", e);
      }
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.encode1("8caW");
      assertTrue(nysiis0.isStrict());
      assertEquals("C", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis(")-ZF*!\u0002'6Jjp}kNy");
      assertEquals("ZFJPNY", string0);
      assertTrue(nysiis0.isStrict());
  }

  @Test(timeout = 4000)
  public void test22()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      Object object0 = nysiis0.encode0("Ybqp-Hd(");
      assertTrue(nysiis0.isStrict());
      assertEquals("YBGFD", object0);
  }

  @Test(timeout = 4000)
  public void test23()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.encode1("By@EbheBevSR p2+n");
      assertEquals("BYABAB", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test24()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      Object object0 = nysiis0.encode0("c}=[ljGyI?xU<h4{Ui");
      assertEquals("CLJGYA", object0);
      assertNotNull(object0);
  }

  @Test(timeout = 4000)
  public void test25()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      String string0 = nysiis0.nysiis("org.apache.commons.codec.EncoderException");
      assertEquals("ORGAPA", string0);
      assertNotNull(string0);
  }

  @Test(timeout = 4000)
  public void test26()  throws Throwable  {
      Nysiis nysiis0 = Nysiis.Nysiis1();
      Object object0 = nysiis0.encode((Object) "s nIO/DScH");
      assertEquals("SNAD", object0);
      assertTrue(nysiis0.isStrict());
  }
}
